packages/core/src/lib/input/__tests__/restricted-text-input.spec.ts

```diff
@@ -0,0 +1,87 @@
+import { describe, it, expect, beforeEach, vi } from 'vitest';
+import { render, screen, waitFor } from '@testing-library/svelte';
+import { RestrictedTextInput } from '$lib';
+import { cxTestArguments, cxTestResults } from '$lib/__tests__/cx-test';
+import userEvent from '@testing-library/user-event';
+import BoundRestrictedTextInput from './restricted-text-input.spec.svelte';
+
+describe('Restricted Text Input', () => {
+  let user: ReturnType<typeof userEvent.setup>;
+
+  beforeEach(() => {
+    user = userEvent.setup();
+  });
+
+  it('renders the input', () => {
+    render(RestrictedTextInput, {
+      value: '',
+      restrictInput: (value: string) => value,
+      tooltipDescription: 'tooltip description',
+      placeholder: 'Enter your name',
+    });
+
+    expect(screen.getByPlaceholderText('Enter your name')).toHaveAttribute(
+      'type',
+      'text'
+    );
+  });
+
+  it('renders with the passed input cx classes', () => {
+    render(RestrictedTextInput, {
+      value: '',
+      restrictInput: (value: string) => value,
+      tooltipDescription: 'tooltip description',
+      placeholder: 'Enter your name',
+      inputCX: cxTestArguments,
+    });
+
+    expect(screen.getByPlaceholderText('Enter your name')).toHaveClass(
+      cxTestResults
+    );
+  });
+
+  it('only allows restricted input', async () => {
+    const { component } = render(BoundRestrictedTextInput, {
+      value: '',
+      restrictInput: (value: string) => value.replaceAll(' ', '-'),
+    });
+
+    const onChange = vi.fn();
+    component.$on('change', onChange);
+
+    const input = screen.getByRole('textbox');
+
+    await user.type(input, 'input with spaces');
+
+    expect(onChange).toHaveBeenCalledWith(
+      expect.objectContaining({ detail: 'input-with-spaces' })
+    );
```

Comment:
Could we add another test that verifies we don't experience the double-dispatch problem of the pre-cleaned value?